app/code/Transoft/Blog/Controller/Adminhtml/Post/Delete.php

```diff
@@ -0,0 +1,76 @@
+<?php
+declare(strict_types=1);
+
+namespace Transoft\Blog\Controller\Adminhtml\Post;
+
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Framework\View\Result\PageFactory;
+use Transoft\Blog\Model\ModelRepository;
+
+/**
+ * Delete post controller
+ */
+class Delete extends Action implements HttpPostActionInterface
+{
+    const ADMIN_RESOURCE = 'Transoft_Blog::blog_manage_posts';
+
+    /**
+     * Index resultPageFactory
+     * @var PageFactory
+     */
+    private $resultPageFactory;
+
+    /**
+     * @var ModelRepository $modelRepository
+     */
+    private $modelRepository;
+
+    /**
+     * Index constructor.
+     * @param Context $context
+     * @param PageFactory $resultPageFactory
+     * @param ModelRepository $modelRepository
+     */
+    public function __construct(
+        Context $context,
+        PageFactory $resultPageFactory,
+        ModelRepository $modelRepository
+    ) {
+        $this->resultPageFactory = $resultPageFactory;
+        $this->modelRepository = $modelRepository;
+        return parent::__construct($context);
+    }
+
+    /**
+     * @inheritDoc
+     *
+     * @throws NoSuchEntityException
+     */
+    public function execute()
+    {
+        $id = (int)$this->getRequest()->getParam('id');
+
+        $post = $this->modelRepository->getById($id);
```

Comment:
getById is throwing an exception for missing blog. Try to delete non-existing blog

Reply:
fixed